stardist/models/base.py

```diff
@@ -70,20 +71,79 @@ def kld(y_true, y_pred):
     return K.mean(K.binary_crossentropy(y_true, y_pred) - K.binary_crossentropy(y_true, y_true), axis=-1)
 
 
+def masked_iou_loss(mask, reg_weight=0, norm_by_mask=True):
+    def iou_loss(y_true, y_pred):
+        axis = -1 if backend_channels_last() else 1 
+
+        inter = K.mean(K.sign(y_pred)*K.square(K.minimum(y_true,y_pred)), axis = axis)
```

Comment:
K.sign?

Reply:
Yes, for negative y_pred (distances) iou should be negative - otherwise one could get arbitrary high iou by simply setting y_pred to very large negative numbers